src/gui/tray/ActivityItemActions.qml

```diff
@@ -0,0 +1,134 @@
+import QtQml 2.15
+import QtQuick 2.15
+import QtQuick.Controls 2.3
+import QtQuick.Layouts 1.2
+import Style 1.0
+
+RowLayout {
+    id: root
+
+    spacing: 20
+
+    property string objectType: ""
+    property variant activityLinks: []
+    property bool displayActions: false
+
+    property color moreActionsButtonColor: "transparent"
+
+    property int maxActionButtons: 0
+
+    property Flickable flickable
+
+    signal triggerAction(int actionIndex)
+
+    function actionButtonIcon(actionIndex, color) {
+        const verb = String(root.activityLinks[actionIndex].verb);
+        const objectType = String(root.objectType);
+        if (verb === "WEB" && (objectType === "chat" || objectType === "call" || objectType === "room")) {
+            return "image://svgimage-custom-color/reply.svg" + "/" + color;
+        }
+
+        return "";
+    }
+
+    function actionButtonText(actionIndex) {
+        const verb = String(root.activityLinks[actionIndex].verb);
+        const objectType = String(root.objectType);
+        if (verb === "DELETE") {
+            return qsTr("Mark as read")
+        } else if (verb === "WEB" && objectType !== "room" && (objectType === "chat" || objectType !== "call")) {
+            return qsTr("Reply")
+        }
+
+        return root.activityLinks[actionIndex].label;
+    }
```

Comment:
This seems like something that should be provided by the repeater's model directly

Reply:
@claucambra  Fixed. I hope that is what you had in mind.